ts/features/pn/components/LegacyServiceCTA.tsx

```diff
@@ -0,0 +1,161 @@
+import { useOnFirstRender } from "../../../utils/hooks/useOnFirstRender";
+import { loadServicePreference } from "../../services/store/actions";
+import {
+  trackPNServiceActivated,
+  trackPNServiceDeactivated,
+  trackPNServiceStartActivation,
+  trackPNServiceStartDeactivation
+} from "../analytics";
+
+type Props = {
+  serviceId: ServiceId;
+  activate?: boolean;
+};
+
+const LoadingIndicator = () => (
+  <ActivityIndicator
+    animating={true}
+    size={"small"}
+    color={IOColors.bluegreyDark}
+    accessible={true}
+    accessibilityHint={I18n.t("global.accessibility.activityIndicator.hint")}
+    accessibilityLabel={I18n.t("global.accessibility.activityIndicator.label")}
+    importantForAccessibility={"no-hide-descendants"}
+  />
+);
+
+const LoadingButton = (props: { isServiceActive: boolean }) => (
+  <ButtonDefaultOpacity
+    block
+    primary
+    style={{
+      backgroundColor: props.isServiceActive
+        ? IOColors.white
+        : IOColors.greyLight,
+      width: "100%"
+    }}
+  >
+    <LoadingIndicator />
+  </ButtonDefaultOpacity>
+);
+
+const ActivateButton = (props: { dispatch: AppDispatch }) => (
+  <ButtonDefaultOpacity
+    block
+    primary
+    onPress={() => {
+      trackPNServiceStartActivation();
+      props.dispatch(pnActivationUpsert.request({ value: true }));
+    }}
+  >
+    <Label color={"white"}>{I18n.t("features.pn.service.activate")}</Label>
+  </ButtonDefaultOpacity>
+);
+
+const DeactivateButton = (props: { dispatch: AppDispatch }) => (
+  <ButtonDefaultOpacity
+    block
+    primary
+    onPress={() => {
+      trackPNServiceStartDeactivation();
+      props.dispatch(pnActivationUpsert.request({ value: false }));
+    }}
+    style={{
+      backgroundColor: IOColors.white
+    }}
+  >
+    <Link weight={"SemiBold"} color={"red"}>
+      {I18n.t("features.pn.service.deactivate")}
+    </Link>
+  </ButtonDefaultOpacity>
+);
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+const LegacyPnServiceCTA = ({ serviceId, activate }: Props) => {
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  const dispatch = useIODispatch();
+  const serviceActivation = useIOSelector(pnActivationSelector);
+  const servicePreference = useIOSelector(servicePreferenceSelector);
+  const servicePreferenceValue = pot.getOrElse(servicePreference, undefined);
+
+  const isLoading =
+    pot.isLoading(servicePreference) ||
+    pot.isLoading(serviceActivation) ||
+    pot.isUpdating(serviceActivation);
+
+  const isServiceActive =
+    servicePreferenceValue &&
+    isServicePreferenceResponseSuccess(servicePreferenceValue) &&
+    servicePreferenceValue.value.inbox;
+
+  useEffect(() => {
+    const wasUpdating = isUpdating;
+    const isStillUpdating = pot.isUpdating(serviceActivation);
+    const isError = pot.isError(serviceActivation);
+    if (wasUpdating && !isStillUpdating) {
+      if (isError) {
+        showToast(I18n.t("features.pn.service.toast.error"), "danger");
```

Comment:
showToast is deprecated 
```suggestion
        IOToast.error(I18n.t("features.pn.service.toast.error"))
```

PS: don't forget to change the import